### ISSUE TITLE
Up sweep in parallel scan

### ISSUE THREAD

**Issue by anhtienng** [alxndrTL/mamba.py#46]
Thank you for your great work.

In your parallel scan, when `Xa.size(2) == 2` or `Xa.size(2) == 1`, why you skip the up-sweep operation ? 
(line 64 in [file](https://github.com/alxndrTL/mamba.py/blob/main/mambapy/pscan.py))

Is it related to the fact `this isn't a static tree but a representation of the evolution of our tensor in memory` in your [document](https://github.com/alxndrTL/mamba.py/blob/main/docs/pscan.ipynb) ?

**Comment by anhtienng** [comment]
I found it by myself.
So when `Xa.size(2) == 2` or `Xa.size(2) == 1`, we don't need `up-sweep` to calculate the cumulative sums, they are already done during `down-sweep`
